backend/services/user_service.py

```
@@ -0,0 +1,515 @@
+            # Get user and mark email as verified
+            user = self.get_user(magic_link.email)
+            if user:
+                self.update_user(
+                    magic_link.email,
+                    email_verified=True,
+                    last_login_at=datetime.utcnow()
+                )
+                user = self.get_user(magic_link.email)  # Refresh
+
+            logger.info(f"Magic link verified for {magic_link.email}")
+            return True, user, "Successfully authenticated"
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)